File: c/hash-table/isomorphic-strings[1].c
bool isIsomorphic(char* s, char* t) {
    int ms[256] = {0};
    int mt[256] = {0};

    int len = strlen(s);

    for(int i = 0; i < len; ++i){
        if(ms[s[i]] != mt[t[i]])
            return false;

        ms[s[i]] = i + 1;
        mt[t[i]] = i + 1;
    }

    return true;
}

File: python/hash-table/isomorphic-strings[1].py
class Solution(object):
    def isIsomorphic(self, s, t):
        ms = {}
        mt = {}

        for i in range(len(s)):
            if ms.get(s[i],-1) != mt.get(t[i],-1):
                return False

            ms[s[i]] = i
            mt[t[i]] = i

        return True
